import "mapreduce/yt/interface/protos/extension.proto";

package NAntiRobot.NCacheSyncProto;

message TRequest {
    required string Request = 1;
    optional string RequesterAddr = 2;
    optional bool IsTraining = 3;
    optional string CacherHost = 4;
    optional int32 RedirectType = 5;
    optional string BlockReason = 6;
    optional bool WasBlocked = 7;
    optional float RobotnessHeaderValue = 8;
    optional float Suspiciousness = 9;
    optional bool CbbWhitelist = 10;
    optional bool SpravkaIgnored = 11;
    optional bool SuspiciousBan = 12;
    optional bool CatboostWhitelist = 13;
    optional string UniqueKey = 14;
    optional bool Degradation = 15;
    repeated uint32 ExperimentsTestId = 16;
    optional float CacherFormulaResult = 17;
    optional bool BanSourceIp = 18;
    optional bool CatboostBan = 19;
    optional fixed64 RandomParam = 20;
    optional bool ApiAutoVersionBan = 21;
    optional bool BanFWSourceIp = 22;
    optional uint64 FirstArrivalTime = 23;
};

message TUid {
    optional uint64 IdLo = 1;
    optional uint64 IdHi = 2;
    optional uint32 Namespace = 3;
    optional uint32 AggrLevel = 4;
}

message TCbbRule {
    optional uint32 Entry = 1;
    optional uint32 Group = 2;
    // optional string Bin = 3;
}

message TCbbBanEntry {
    repeated TCbbRule Rules = 1 [(NYT.flags) = SERIALIZATION_YT];
    optional uint64 ExpirationTime = 2;
}

message TBlockAction {
    required bytes Data = 1;
};

message TBanAction {
    // required bytes Data = 1;
    optional TUid Uid = 2 [(NYT.flags) = SERIALIZATION_YT];
    optional uint32 HostType = 3;
    optional uint64 MatrixnetExpirationTime = 4;
    // optional uint64 YqlExpirationTime = 5;
    // optional uint64 CbbExpirationTime = 6;
    repeated TCbbBanEntry YqlEntries = 7 [(NYT.flags) = SERIALIZATION_YT];
    repeated TCbbBanEntry CbbEntries = 8 [(NYT.flags) = SERIALIZATION_YT];
};

message TMessage {
    optional TRequest Request = 1;
    repeated TBlockAction Blocks = 2;
    repeated TBanAction Bans = 3;
};

message TCaptchaInput {
    required TRequest Request = 1;
    required bool IsCorrect = 2;
};
